e2e/terraform/shared/config/provision-client.sh

```diff
@@ -22,11 +22,26 @@ sleep 10
 # Add hostname to /etc/hosts
 echo "127.0.0.1 $(hostname)" | sudo tee --append /etc/hosts
 
-# Add Docker bridge network IP to /etc/resolv.conf (at the top)
+# Use dnsmasq first and then docker bridge network for DNS resolution
 DOCKER_BRIDGE_IP_ADDRESS=$(/usr/local/bin/sockaddr eval 'GetInterfaceIP "docker0"')
-echo "nameserver $DOCKER_BRIDGE_IP_ADDRESS" | sudo tee /etc/resolv.conf.new
-cat /etc/resolv.conf | sudo tee --append /etc/resolv.conf.new
-sudo mv /etc/resolv.conf.new /etc/resolv.conf
+{
+    echo "nameserver 127.0.0.1"
+    echo "nameserver $DOCKER_BRIDGE_IP_ADDRESS"
+} > /tmp/resolv.conf
```

Comment:
Very nit-picky personal thing - I find it nicer to use heredoc for these situations or `printf`

```suggestion
cat <<EOF > /tmp/resolv.conf
nameserver 127.0.0.1
nameserver $DOCKER_BRIDGE_IP_ADDRESS
EOF
```

Reply:
Works for me. Reads nicer.